### PR TITLE
refactor: Images layout into functional component

### PR DESCRIPTION
**Pull request by kwajiehao**
This PR refactors the `Images` layout into a functional component

### CODE REVIEW

File: src/layouts/Images.jsx [hunk 1]
```diff
@@ -10,53 +10,48 @@ import MediaUploadCard from '../components/media/MediaUploadCard';
...
-      });
-      const { images } = resp.data;
-      if (this._isMounted) this.setState({ images });
-    } catch (err) {
-      console.log(err);
+  useEffect(() => {
+    let _isMounted = true
+    const fetchData = async () => {
+      try {
+        const resp = await axios.get(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/images`, {
```
Comment (gweiying) [inline src/layouts/Images.jsx:22]: Should we abstract this to the `api.js` file?

**Review by gweiying — COMMENTED** [review]
In general looks good to me, just two small nits to consider:

1) We can consider abstracting the axios GET request to the `api.js` file
2) Adding a success toast or error toast after completed rename action

**Comment by kwajiehao** [comment]
> In general looks good to me, just a small nit to consider:
> 
> 1. We can consider abstracting the axios GET request to the `api.js` file
>    ~2) Adding a success toast or error toast after completed rename action~ opened in #365

Abstracting the axios GET request (together with the shift to use `react-query` library to retrieve data) is done in a subsequent PR based off of this (#370) because I wasn't sure that it fell under the scope of refactoring from class -> functional component

**Review by gweiying — APPROVED** [review]
lgtm

**Comment by kwajiehao** [comment]
Note: this PR was reverted by force pushing the `staging` branch back to commit `0d76fa2946db96aa4069e005cfb338c57de6e74e`. This is because there was a bug in setting state for pending image upload:

```
setPendingImageUpload({
        pendingImageUpload: {
          fileName: imageName,
          path: `images%2F${imageName}`,
          content: imageContent,
        }
})
``` 

should have read

```
setPendingImageUpload({
          fileName: imageName,
          path: `images%2F${imageName}`,
          content: imageContent,
})
```

This will be fixed in PR #370, which will also include the changes from this PR

**Pull request merged**